store/s3store.go

```diff
@@ -58,18 +55,8 @@ type S3Store struct {
 	bucket string
 }
 
-// Deprecated; use NewS3StoreWithBucket instead
-func NewS3Store(numRetries int) (*S3Store, error) {
```

Comment:
this feels like a sneak in that isn't necessarily related to supporting contexts - does this deserve its own PR/changelog line item for the v3 release?

Reply:
From the integrator standpoint, yes. Let me spin off the removal into a separate PR that we can merge before this one.

It's related because removing it avoids needing to add `ctx` to it. The constructor has been deprecated for over five years.

Reply:
https://github.com/segmentio/chamber/pull/502